### PR TITLE
fix: call loadUrl when opening new windows from links

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Closes https://github.com/electron/electron/issues/34152.

Fixes an issue where windows opened from new links wouldn't properly load URLs. This happened as a result of cleanup in https://github.com/electron/electron/commit/d44a187d0b226800fe0cb4f7a0d2b36c871b27cd - a section of code was removed which loaded the url for non-guest windows in `openGuestWindow`. It's actually the case that this code is necessary when we're opening new windows from links as well. When we open a new window from a link via `Shift+LeftClick`, this routes through `OpenUrlFromTab`, at which point the browser process is responsible for initiating navigation in the new window. This PR restores that code and updates the comment accordingly.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Fixes an issue where some `BrowserWindow`s opened from new links wouldn't properly load URLs.

### CODE REVIEW

File: spec-main/fixtures/apps/open-new-window-from-link/main.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    wc.on('did-finish-load', () => {
+      const title = wc.getTitle();
+      if (title === 'Window From Link') {
+        process.exit(0);
+      }
+    });
+  });
+
+  createWindow();
+});
```
Comment (samuelmaddock) [inline spec-main/fixtures/apps/open-new-window-from-link/main.js:64]: Do the apps added under this directory get run as part of the test suite?

**Review by samuelmaddock — COMMENTED** [review]
lgtm, just had a question about the test

**Review by VerteDinde — APPROVED** [review]

File: spec-main/fixtures/apps/open-new-window-from-link/main.js (hunk 1 above)
Comment (codebytere) [reply spec-main/fixtures/apps/open-new-window-from-link/main.js:64]: oops i apparently forgot to push that 🙃 - fixed.

**Review by samuelmaddock — APPROVED** [review]

**Review by MarshallOfSound — APPROVED** [review]

**Comment by pushkin-** [comment]
Is this going to be backported to Electron 18?

**Pull request merged**